### PR TITLE
Core dependency and publish updates

### PR DESCRIPTION
**Pull request by sinha108**
## Description

This PR contains the following updates:
- download evosuite, randoop, and crawljax jars from jitpack (for crawljax, dependencies are specified on the core and specific plugins instead of one fat jar with all dependencies)
- disable core unit/ui jar publish on github packages
- disable use of github token in CI workflows
- related code updates for references to jars

Related to # (issue)

## Type of Change

Please check the types of changes your PR introduces.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Refactoring (non-breaking code restructuring that preserves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Build-related update (CI workflow, test cases)
- [ ] Documentation update
- [ ] Other (please describe):

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so that it can be replicated.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tackle-test-generator-unit/pom.xml [hunk 1]
```diff
@@ -35,8 +29,8 @@
     <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
     <maven-javadoc-plugin.version>3.0.0</maven-javadoc-plugin.version>
     <coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>
-	<evosuite.version>1.2.0</evosuite.version>
-    <randoop.version>4.3.0</randoop.version>
+	  <evosuite.version>v1.2.0</evosuite.version>
```
Comment (rachelt44) [inline tackle-test-generator-unit/pom.xml:32]: why added v to the version numbers for randoop and evosuite?

File: tackle-test-generator-unit/pom.xml [hunk 2]
```diff
@@ -89,19 +82,15 @@
       <version>[4.5.13,)</version>
     </dependency>
     <dependency>
-      <groupId>org.evosuite</groupId>
-	  <artifactId>evosuite</artifactId>
+      <groupId>com.github.EvoSuite.evosuite</groupId>
+      <artifactId>evosuite-master</artifactId>
```
Comment (rachelt44) [inline tackle-test-generator-unit/pom.xml:86]: why do we need both evosuite and evosuite-master?

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java [hunk 3]
```diff
@@ -164,8 +164,8 @@ public static class AppUnderTest {
         public String appPath;
         public String appClasspathFilename;
         public String testPlanFilename;
-
```
Comment (rachelt44) [inline tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java:167]: all empty lines are marked as changed - any idea why?

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java [hunk 4]
```diff
@@ -917,12 +917,12 @@ public IntegrationAppUnderTest(String appName,
...
             this.testSeqFilename = testSeqFilename;
             this.targetClassList = targetClassList;
-            
+
             if (this.testSeqFilename == null) {
                 this.testSeqFilename = appName + "_" + EvoSuiteTestGenerator.class.getSimpleName() + "_" +
-                    Constants.INITIALIZER_OUTPUT_FILE_NAME_SUFFIX + "," +
-                    appName + "_" + RandoopTestGenerator.class.getSimpleName() + "_" +
                     Constants.INITIALIZER_OUTPUT_FILE_NAME_SUFFIX;
+//                    + "," + appName + "_" + RandoopTestGenerator.class.getSimpleName() + "_" +
```
Comment (rachelt44) [inline tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java:924]: The flaky tests are in the cli regression, not in the core regression. Please revert this change.

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java [hunk 5]
```diff
@@ -117,7 +118,7 @@ public void testModelerInitializerExtender() throws Exception {
             // create building-block test sequences using combined test generator
             TestSequenceInitializer seqInitializer = new TestSequenceInitializer(testApp.appName,
                 testApp.testPlanFilename, testApp.appPath, testApp.appClasspathFilename,
-                Constants.COMBINED_TEST_GENERATOR_NAME, 5, false, false, "");
```
Comment (rachelt44) [inline tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java:120]: The flaky tests are in the cli regression, not in the core regression. Please revert this change.

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java [hunk 6]
```diff
@@ -56,7 +56,7 @@ public void testGenerateTests() throws Exception {
         // assert that input/output files for randoop are created
         //assertTrue(new File(RandoopTestGenerator.RANDOOP_CLASSLIST_FILE_NAME).exists());
         //assertTrue(new File(RandoopTestGenerator.RANDOOP_METHODLIST_FILE_NAME).exists());
-        assertTrue(new File("DayTrader"+RandoopTestGenerator.RANDOOP_OUTPUT_DIR_NAME_SUFFIX).exists());
+//        assertTrue(new File("DayTrader"+RandoopTestGenerator.RANDOOP_OUTPUT_DIR_NAME_SUFFIX).exists());
```
Comment (rachelt44) [inline tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java:59]: The flaky tests are in the cli regression, not in the core regression. Please revert this change.

**Review by rachelt44 — CHANGES_REQUESTED** [review]
Left inline comments

File: tackle-test-generator-unit/pom.xml (hunk 1 above)
Comment (sinha108) [reply tackle-test-generator-unit/pom.xml:32]: @rachelt44: That's how evosuite and randoop are available on jitpack:
- https://jitpack.io/#evosuite/evosuite/v1.2.0
- https://jitpack.io/#randoop/randoop/v4.3.0

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java (hunk 3 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java:167]: The IDE apparently removed tabs, and in many cases replaces tabs with spaces, on opening the files

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java (hunk 4 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java:924]: This test was also failing consistently (I tried 3-4 runs); I can revert it if you'd prefer that

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java (hunk 5 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java:120]: This test was also failing consistently (I tried 3-4 runs); I can revert it if you'd prefer that

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java (hunk 6 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java:59]: This test was also failing consistently (I tried 3-4 runs); I can revert it if you'd prefer that

File: tackle-test-generator-unit/pom.xml (hunk 2 above)
Comment (sinha108) [reply tackle-test-generator-unit/pom.xml:86]: Yes, I know; that's what's available on jitpack (I spent a lot of time figuring this out)

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java (hunk 6 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/core/RandoopTestGeneratorTest.java]: Reverted

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java (hunk 5 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestgenIntegrationTest.java]: Reverted

File: tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java (hunk 4 above)
Comment (sinha108) [reply tackle-test-generator-unit/test/java/org/konveyor/tackle/testgen/TestUtils.java]: Reverted

**Review by rachelt44 — APPROVED** [review]

**Pull request merged**